packaging_automation/tests/test_package_cloud_statistics_collector.py

```diff
@@ -20,15 +24,22 @@
 def test_fetch_and_save_package_cloud_stats():
     db = create_engine(db_connection_string(db_params=db_parameters, is_test=True))
     db.execute(text(f'DROP TABLE IF EXISTS {PackageCloudDownloadStats.__tablename__}'))
-    pack_count = package_count(ORGANIZATION, REPO, PACKAGE_CLOUD_API_TOKEN)
     session = db_session(db_params=db_parameters, is_test=True)
     page_record_count = 3
     parallel_count = 3
+
+    result = stat_get_request(
+        package_list_with_pagination_request_address(PACKAGE_CLOUD_API_TOKEN, 1, ORGANIZATION, REPO, 100),
```

Comment:
A bit confused here. Doesn't this only look at the first page of packages? But below we will go through all of the pages.

Reply:
After adding filters getting package count from directly repo became useless, since packages other than debug, pgaf-cli and citus-ha needs to be calculated. So In line 30 , I get all the packages in the repo (I'm using azure as test repo and package count is 24 which is lower than 100) .

Reply:
refactored and create a method which is responsible of getting the filtered reference package count